internal/blocktx/processor.go

```diff
@@ -426,206 +449,241 @@ func (p *Processor) processBlock(msg *p2p.BlockMessage) (err error) {
 		tracing.EndTracing(span, err)
 	}()
 
-	timeStart := time.Now()
-
 	blockHash := msg.Header.BlockHash()
-	previousBlockHash := msg.Header.PrevBlock
-	merkleRoot := msg.Header.MerkleRoot
+	blockHeight := msg.Height
 
-	// don't process block that was already processed
+	p.logger.Info("processing incoming block", slog.String("hash", blockHash.String()), slog.Uint64("height", blockHeight))
+
+	// check if we've already processed that block
 	existingBlock, _ := p.store.GetBlock(ctx, &blockHash)
```

Comment:
I think we shouldn't return non-processed blocks by default from any repository method.
IMO situation when we need non-processed block is exceptional and should be another method for this.

Notice that this change will simplify logic in few places

Reply:
Good point, done and removed the logic